### PR TITLE
fix: finding and updating transaction errors in confirmation pages state

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
The PR fixes the issue - whenever loading balance is delayed, the Warning for needing more ETH is displayed and cannot perform the tx once the balance is loaded.
The reason for this issue was wrong state update for error, I fixed this but re-calculating error every time transaction / user balance / contracts balance changes. Thus as soon as balance is fetched the error goes away.

## **Manual testing steps**
1. Go to test-dapp
2. Simulate a slow response from the getBalance request
3. Click on Send EIP1559
4. See how warning for needing more ETH appears and Confirm button is disabled
5. Wait until correct balance is loaded
6. See how warning is removed and Confirm button is still enabled

## **Related issues**

_Fixes #6535

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/TransactionEditor/index.js [hunk 1]
```diff
@@ -136,6 +136,7 @@ class TransactionEditor extends PureComponent {
     toFocused: false,
     ensRecipient: undefined,
     ready: false,
+    error: true,
```
Comment (blackdevelopa) [inline app/components/UI/TransactionEditor/index.js:139]: Why do we set the error default state to true? 🤔

Comment (jpuri) [reply app/components/UI/TransactionEditor/index.js:139]: This is to ensure that confirm button is disabled till user balance is loaded and verified.

Comment (jpuri) [reply app/components/UI/TransactionEditor/index.js]: I updated error to default to `undefined`.

Comment (jpuri) [reply app/components/UI/TransactionEditor/index.js]: I have reset the value to `true` but added comment for it.

I found that it is required to ensure correct behaviour for other components which are using `<TransactionReview` child component.

**Review by blackdevelopa — DISMISSED** [review]

**Comment by seaona** [comment]
I've started a Bitrise run here, to check e2e when finishes: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e811b91a-e6f0-4746-85c5-11e77f602fa3

**Review by blackdevelopa — APPROVED** [review]

**Review by segun — APPROVED** [review]

File: app/components/UI/TransactionReview/index.js [hunk 2]
```diff
@@ -191,6 +187,10 @@ class TransactionReview extends PureComponent {
      * ETH or fiat, depending on user setting
      */
     primaryCurrency: PropTypes.string,
+    /**
+     * Error blockaid transaction execution, undefined value signifies no error.
+     */
+    error: PropTypes.oneOf[(PropTypes.bool, PropTypes.string)],
```
Comment (wachunei) [inline app/components/UI/TransactionReview/index.js:193]: This is a typo, `PropTypes.oneOf` is a function that takes an array as argument:

```js
    error: PropTypes.oneOf([PropTypes.bool, PropTypes.string]),
```

No impact other than development env proptypes check.

Comment (wachunei) [reply app/components/UI/TransactionReview/index.js:193]: @jpuri @blackdevelopa @segun pinging this for your visibility

Comment (wachunei) [reply app/components/UI/TransactionReview/index.js:193]: Fixed by https://github.com/MetaMask/metamask-mobile/pull/7821

**Pull request merged**